### ISSUE TITLE
Undefined Symbol gen_parsing_ops_py_wrappers_cc when building pip package

### ISSUE THREAD

**Issue by gbolcer** [tensorflow/tensorflow#15958]
### System information

greg@salt:~/code/tensorflow$ more tf_env.txt 

== cat /etc/issue ===============================================
Linux salt 4.13.0-21-generic #24-Ubuntu SMP Mon Dec 18 17:29:16 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
VERSION="17.10 (Artful Aardvark)"
VERSION_ID="17.10"
VERSION_CODENAME=artful

== are we in docker =============================================
No

== compiler =====================================================
(configure set to gcc-6)
c++ (Ubuntu 7.2.0-8ubuntu3) 7.2.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== uname -a =====================================================
Linux salt 4.13.0-21-generic #24-Ubuntu SMP Mon Dec 18 17:29:16 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

== check pips ===================================================
numpy (1.14.0)
protobuf (3.5.1)
s2clientprotocol (1.1)
tensorflow-gpu (1.4.0)
tensorflow-serving-api (1.4.0)
tensorflow-tensorboard (0.4.0rc3)

== check for virtualenv =========================================
False

== tensorflow import ============================================
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "tensorflow/python/__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "tensorflow/python/pywrap_tensorflow.py", line 25, in <module>
    from tensorflow.python.platform import self_check
ImportError: No module named platform

== env ==========================================================
LD_LIBRARY_PATH /usr/local/cuda/lib64:/opt/tensorlibs/lib
DYLD_LIBRARY_PATH /usr/local/cuda/lib64:/opt/tensorlibs/lib

== nvidia-smi ===================================================
Mon Jan  8 13:25:47 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 387.26                 Driver Version: 387.26                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1070    Off  | 00000000:01:00.0  On |                  N/A |
| 24%   34C    P8     9W / 151W |    596MiB /  8112MiB |      3%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0       988      G   /usr/lib/xorg/Xorg                           373MiB |
|    0      1384      G   /usr/bin/compiz                              116MiB |
|    0      8602      G   ...-token=09D12A9300046A211380F83BAD0A60C6    46MiB |
|    0     15983      G   ...-token=29A09CA037A6A6610BD563090AA5B5DA    56MiB |
+-----------------------------------------------------------------------------+

== cuda libs  ===================================================
/usr/local/cuda-8.0/doc/man/man7/libcudart.so.7
/usr/local/cuda-8.0/doc/man/man7/libcudart.7
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudart_static.a
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudart.so.8.0.61
/usr/local/cuda-9.1/doc/man/man7/libcudart.so.7
/usr/local/cuda-9.1/doc/man/man7/libcudart.7
/usr/local/cuda-9.1/targets/x86_64-linux/lib/libcudart_static.a
/usr/local/cuda-9.1/targets/x86_64-linux/lib/libcudart.so.9.1.85
/usr/local/cuda-9.0/doc/man/man7/libcudart.so.7
/usr/local/cuda-9.0/doc/man/man7/libcudart.7
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudart_static.a
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudart.so.9.0.103

### Describe the problem
Build command below fails w/ undefined symbol.  
bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package --verbose_failures

ERROR: /home/greg/code/tensorflow/tensorflow/python/BUILD:1398:1: Executing genrule //tensorflow/python:parsing_ops_pygenrule failed (Exit 127): bash failed: error executing command 
  (cd /home/greg/.cache/bazel/_bazel_greg/ca148a14f6c24000015970c3c0a435f7/execroot/org_tensorflow && \
  exec env - \
    CUDA_TOOLKIT_PATH=/usr/local/cuda \
    CUDNN_INSTALL_PATH=/usr/local/cuda-9.1 \
    GCC_HOST_COMPILER_PATH=/usr/bin/gcc-6 \
    LD_LIBRARY_PATH=/usr/local/cuda/lib64:/opt/tensorlibs/lib \
    PATH=/opt/ant/bin:.:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/lib/jvm/java-9-oracle/bin:/usr/lib/jvm/java-9-oracle/db/bin:/opt/gephi/bin:/opt/eclipse:/opt/gradle/bin \
    PYTHON_BIN_PATH=/usr/bin/python3 \
    PYTHON_LIB_PATH=/usr/lib/python3/dist-packages \
    TF_CUDA_CLANG=0 \
    TF_CUDA_COMPUTE_CAPABILITIES=6.1 \
    TF_CUDA_VERSION=9.1 \
    TF_CUDNN_VERSION=7 \
    TF_NEED_CUDA=1 \
    TF_NEED_OPENCL_SYCL=0 \
  /bin/bash bazel-out/k8-py3-opt/genfiles/tensorflow/python/parsing_ops_pygenrule.genrule_script.sh)
bazel-out/host/bin/tensorflow/python/gen_parsing_ops_py_wrappers_cc: symbol lookup error: bazel-out/host/bin/tensorflow/python/gen_parsing_ops_py_wrappers_cc: undefined symbol: _ZN10tensorflow17ParseExampleAttrs10FinishInitEv
Target //tensorflow/tools/pip_package:build_pip_package failed to build
INFO: Elapsed time: 63.574s, Critical Path: 19.07s
FAILED: Build did NOT complete successfully

### Source code / logs
greg@salt:~/code/tensorflow$ dpkg -l | grep cuda
ii  cuda                                       9.1.85-1                                    amd64        CUDA meta-package
ii  cuda-9-1                                   9.1.85-1                                    amd64        CUDA 9.1 meta-package
....
ii  libcudnn7                                  7.0.5.15-1+cuda9.1                          amd64        cuDNN runtime libraries
ii  libcudnn7-dev                              7.0.5.15-1+cuda9.1                          amd64        cuDNN development libraries and headers
ii  libcudnn7-doc                              7.0.5.15-1+cuda9.1                          amd64        cuDNN documents and samples

**Comment by cy89** [comment]
@martinwicke @javiribera @jart PTAL.

**Comment by gbolcer** [comment]
Apologies, this was a false alarm.  After posting, I noticed the /opt/tensorlibs which was an old directory where I used static libraries to get around a previous syntaxnet gcc linking error.  I also had tensorflow 1.5 and tensorflow-gpu 1.4.1 both installed.  I removed that /opt/tensorlibs directori, did a pip3 uninstall for both tensorflow and tensorflow-gup,  and did a bazel clean.  That solved the issue.
